Build operator-sdk failed

## Bug Report

<!-- 
Thanks for filing an issue! Before hitting the button, please answer these questions.

Fill in as much of the template below as you can. If you leave out information, we can't help you as well.
-->

**What did you do?**
I build the operator-sdk per this [quick-start](https://github.com/operator-framework/operator-sdk#quick-start), but got errors when running `make dep`:
```go
mac:operator-sdk jianzhang$ pwd
/Users/jianzhang/goproject/src/github.com/operator-framework/operator-sdk
mac:operator-sdk jianzhang$ git branch
* master
mac:operator-sdk jianzhang$ make dep
# Bringing vendor into sync
(1/38) Wrote github.com/Masterminds/semver@v1.4.2: missing from vendor
grouped write of manifest, lock and vendor: failed to export google.golang.org/appengine: fatal: failed to unpack tree object 4a4468ece617fc8205e99368fa2200e9d1fad421
: exit status 128
make: *** [dep] Error 1
```

So, I tried to update the Golang packages, got errors:
```go
mac:operator-sdk jianzhang$ go get -u ./...
package k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource: cannot find package "k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource" in any of:
	/usr/local/Cellar/go/1.11.1/libexec/src/k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource (from $GOROOT)
	/Users/jianzhang/goproject/src/k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource (from $GOPATH)
```

**What did you expect to see?**
Build the operator-sdk successfully.

**What did you see instead? Under which circumstances?**


**Environment**
* operator-sdk version:
master branch
```console
mac:operator-sdk jianzhang$ git log
commit 3a20d44134fc248af7f41e700ba5232c264b1707 (HEAD -> master, origin/master, origin/HEAD)
Author: Matt Dorn <madorn@gmail.com>
Date:   Thu Dec 6 16:26:24 2018 -0600

    add cluster-scoped flag to v0.2.0 changelog (#822)
    
    add cluster-scoped flag to v0.2.0 changelog
```

* Kubernetes version information:
Using the `oc` instead.
mac:operator-sdk jianzhang$ oc version
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth

* Kubernetes cluster kind: 

* Are you writing your operator in ansible or go?
go

**Possible Solution**
<!--- Only if you have suggestions on a fix for the bug -->

**Additional context**
```console
mac:operator-sdk jianzhang$ dep version
dep:
 version     : v0.5.0
 build date  : 2018-07-26
 git hash    : 224a564
 go version  : go1.10.3
 go compiler : gc
 platform    : darwin/amd64
 features    : ImportDuringSolve=false
mac:operator-sdk jianzhang$ echo $GOPATH
/Users/jianzhang/goproject
mac:operator-sdk jianzhang$ echo $GOROOT
/usr/local/Cellar/go/1.11.1/libexec
mac:operator-sdk jianzhang$ go version
go version go1.11.1 darwin/amd64
```

I am able to build operator-sdk (just tried on recent master) but the operator build is failing for me. 
```
cd $GOPATH/src/github.com/example-inc/
operator-sdk new app-operator
cd app-operator
operator-sdk add api --api-version=app.example.com/v1alpha1 --kind=AppService
operator-sdk add controller --api-version=app.example.com/v1alpha1 --kind=AppService
operator-sdk build quay.io/example/app-operator
```

leads to:
```
# github.com/example-inc/app-operator/vendor/sigs.k8s.io/controller-runtime/pkg/client
vendor/sigs.k8s.io/controller-runtime/pkg/client/unstructured_client.go:49:24: undefined: v1.CreateOptions
vendor/sigs.k8s.io/controller-runtime/pkg/client/unstructured_client.go:67:24: undefined: v1.UpdateOptions
vendor/sigs.k8s.io/controller-runtime/pkg/client/unstructured_client.go:145:26: too many arguments in call to r.UpdateStatus
vendor/sigs.k8s.io/controller-runtime/pkg/client/unstructured_client.go:145:30: undefined: v1.UpdateOptions
```

Any idea, please?

--- a/Makefile
+++ b/Makefile
@@ -24,6 +24,9 @@ format:
 dep:
 	$(Q)dep ensure -v
 
+dep-update:
+	$(Q)dep ensure -update -v
+
 clean:
 	$(Q)rm -rf build
 

